src/client/app/helpers/trusted.snippet.helper.ts

```diff
@@ -0,0 +1,26 @@
+class TrustedSnippet {
+    private _trustedSnippetsKey = 'trusted_snippets';
+
+    isSnippetTrusted(snippetId: string, gistId: string): boolean {
+        let trustedSnippets = JSON.parse(window.localStorage.getItem(this._trustedSnippetsKey));
+        if (!trustedSnippets) {
+            trustedSnippets = {};
+        }
+        /* Samples are automatically trusted. Check local storage for gists. */
+        if (!gistId || trustedSnippets[snippetId]) {
```

Comment:
For performance you should check gistId first thing in this function and return true. This saves the time it takes the browser to look up the object in localStorage and parse it.

Reply:
Added.